### ISSUE TITLE
Add python3 support?

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#28]
Hello,

When trying to install python-keyring, I got the following error:

```
Traceback (most recent call last):
  File "setup.py", line 12, in <module>
    from extensions import get_extensions
  File "/home/victor/code/pkgbuild/python/python-keyring/src/keyring-0.4/extensions.py", line 9, in <module>
    import commands
ImportError: No module named commands
```

I think this is linked to this:
http://docs.python.org/library/commands.html

In particular to: "Deprecated since version 2.6: The commands module has been removed in Python 3.0. Use the subprocess module instead."
Could we find a good solution to work with both versions?
Thanks!

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/28
- Originally reported by: Anonymous
- Originally created at: 2010-10-20T10:08:48.723

**Comment by jaraco** [comment]
Good idea. Definitely needs to be done. If I can find some time, I'll take a stab at it.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
What are your plans for Python3 support? Should the code run under both Python 2 and 3 at the same time? If so, the minimal supported Python 2 version would be Python 2.6...

The commands can easily replace with the subprocess module, an easy task to get one's feet wet.

---

Original comment by: Marek Kubica

**Comment by jaraco** [comment]
Actually, what is the minimal supported version of Python?

---

Original comment by: Jonathan Ballet

**Comment by jaraco** [comment]
I believe the minimal supported version is 2.4. I do see 2.3-incompatible syntax, but I think Python 2.4 is supported. I've documented this in the setup.py using trove classifiers.

I've also made some progress on Python 3 support on Windows. Currently the package is working for me on Windows under both Python 2.7 and 3.2 and all unit tests that are not skipped are passing.

I'm using one codebase using distribute's use_2to3 support and a few inline compatibility shims. So far, I've been able to maintain Python 2.4 compatibility (I think).

I hope later this week to work on support for Unix platforms.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
I successfully ran the tests on Linux under Python 3.2, although I did not have KDE or Gnome support. As far as I can tell, there is no gnomekeyring or PyKDE4 for Python 3. Additionally, I wouldn't know the first thing about getting a KDE environment to test the KDE keyring.

However, it is safe to say that the other keyrings are running fine under Python 3, and based on a cursory review of the code, I don't see anything that would suggest that getting those keyrings running on Python 3 would be any problem, assuming the backend support is there.

With that, I'm marking this ticket as closed. If specific issues arise with keyring under Python 3, they should be filed as separate tickets.

---

Original comment by: Jason R. Coombs
